### ISSUE TITLE
Nested each broken in 2.0.0

### ISSUE THREAD

**Issue by Keruspe** [sunng87/handlebars-rust#275]
I have some template that look like this:

```
{{#each protocol.classes as |class| ~}}
{{#each class.methods as |method| ~}}
{{method.id}}
{{/each ~}}
{{/each ~}}
```

With data that look like this (passed as "protocol" to the template):

```
{
  "classes": [
    {
      "methods": [
        "id": 1
      ]
    }
  ]
}
```

This works fine with handlebars 2.0.0-beta.2 but fails with 2.0.0

**Comment by Keruspe** [comment]
You can pull https://github.com/Keruspe/handlebars-rust/commit/bdd58fb1d88fe4db589f1037220fd3500f886252 for a failing test reproducing it.
If you cherry-pick the test on top of 2.0 beta.2, it passes, on top of 2.0 final it doesn't

**Comment by Keruspe** [comment]
2.0 beta.3 fails too, fwiw

**Comment by sunng87** [comment]
Seems to be a regression issue from #263, when block param used in  nested each. Thanks for reporting.

**Comment by sunng87** [comment]
@Keruspe just update I have been working on a fix these days. The root cause is when the parameter of `each` contains a reference starts with block parameter, there is an error in base path settings. 

My initial fix is causing benchmark dropping a bit so I will be working on a new solution this week. Also this may bring in API changes for `context.rs` and probably to be released in `3.0.0`

**Comment by Keruspe** [comment]
Thanks !

Any chance of getting a release with this fix?

**Comment by sunng87** [comment]
The fix is released in 2.0.1
